src/events/MouseEvents.hxx

```diff
@@ -0,0 +1,58 @@
+#ifndef MOUSE_EVENTS_HXX
+#define MOUSE_EVENTS_HXX
+
+#include "../util/Meta.hxx"
+
+/**
+ * @brief A change in the mouse (or device)'s position
+ */
+struct MousePositionEvent
+{
+  int xPosition;
+  int yPosition;
+  int xVelocity;
+  int yVelocity;
+};
+
+/**
+ * @brief A click interaction
+ */
+struct ClickEvent
+{
+  int32_t xPosition;
+  int32_t yPosition;
+  /**
+   * @brief One-hot encoding for the state of the mouse or device
+   * @details Exactly one of {Left, Right, Middle} must be set 
+   *          and exactly one of {Pressed, Released} must be set
+   */
+  enum MouseButtonID : uint8_t
+  {
+    Left =      1 << 1,
+    Right =     1 << 2,
+    Middle =    1 << 3
+  };
+  enum MouseButtonState : uint8_t
+  {
+    Pressed =   1 << 4,
+    Released =  1 << 5
```

Comment:
Everything is shifted 1 bit more to the left. Why the second enumeration starts from 5th bit but not 1st?